sql/2019/10_SEO/10_09_a.sql

```diff
@@ -0,0 +1,42 @@
+#standardSQL
+
+# Metric: Content - looking at word count, thin pages, header usage, alt attributes images
+
+CREATE TEMPORARY FUNCTION parseWords(payload STRING, element STRING)
+RETURNS INT64 LANGUAGE js AS '''
+  try {
+    var $ = JSON.parse(payload);
+    var almanac = JSON.parse($._almanac);
+    return almanac['seo-words'][element];
+  } catch (e) {
+    return 0;
+  }
+''';
+
+CREATE TEMPORARY FUNCTION parseTitle(payload STRING, element STRING)
+RETURNS INT64 LANGUAGE js AS '''
+  try {
+    var $ = JSON.parse(payload);
+    var almanac = JSON.parse($._almanac);
+    return almanac['seo-titles'][element];
+  } catch (e) {
+    return 0;
+  }
+''';
+
+
+SELECT
+    # words
+    APPROX_QUANTILES(parseWords(payload,'wordsCount'), 1000)[OFFSET(250)] as quantP25WordsCount,
```

Comment:
Calling parseWords/parseTitle multiple times per URL seems to be slowing the query down. In my other comment I've added an alternate version without the UDFs that uses an inner query to do the JSON extraction once for each of the 4 metrics.